app/components/post_list/date_header/date_header.js

```diff
@@ -33,7 +34,7 @@ export default class DateHeader extends PureComponent {
         };
 
         if (timeZone) {
-            dateFormatProps.timeZone = timeZone;
+            dateFormatProps.value = new Date(moment.tz(date, timeZone).toDate());
```

Comment:
I think `toDate` returns a `Date` object, so we don't need to wrap it in another one with `new Date`
```suggestion
            dateFormatProps.value = moment.tz(date, timeZone).toDate();
```